### PR TITLE
Refactor ObservationRepository

### LINKED ISSUE #523: [Code health] Move creation of observation mutation to ObservationRepository
`EditObsevationViewModel` has a lot of logic and it would be better if we move the logic for creating observation mutations in the `ObservationRepository` where the same is done for creating observations.

This would help in simplifying the logic class and increasing readability.

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Closes #523 

Move boilerplate code for generating mutations to ObservationRepository

### CODE REVIEW

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java [hunk 1]
```diff
@@ -303,21 +293,13 @@ private void onObservationLoaded(Observation observation) {
...
-            .setUserId(authManager.getCurrentUser().getId())
-            .build();
+    if (originalObservation == null) {
+      return Single.error(new IllegalStateException("Observation is null"));
+    }
+
     return observationRepository
-        .applyAndEnqueue(observationMutation)
+        .addObservationMutation(originalObservation, getResponseDeltas(), isNew)
+        .doOnSubscribe(disposable -> savingProgressVisibility.setValue(View.VISIBLE))
```
Comment (dturner) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:302]: Just an observation, the viewmodel's `saveProgressVisibility` property could be refactored to `isSaving` and the visibility logic could then be handled in the layout XML using databinding e.g. `android:visibility="@{viewmodel.isSaving ? View.VISIBLE : View.GONE}"`

Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:302]: Thanks for the comment! We are already using databinding for toggling the visibility. The only difference is, currently the binded object is an integer type whose value is equal to one of the visibility values. And in this scenario, we would use a boolean type instead with the suggested refactor. Any particular reason for using boolean over integer type?

Comment (dturner) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:302]: Yeah, it's really only a minor thing. The reason for using boolean over integer is that it represents the 2 saving states: "saving" and "not saving". 

Currently the `savingProgressVisibility` integer maps to the 3 possible visibility states: visible, invisible and gone. We don't seem to use "invisible" so refactoring to `isSaving.setValue(true)` makes the code (albeit only slightly) more readable.

Comment (dturner) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:302]: I did the refactor on `ObservationListViewModel` - see what you think :) https://github.com/shobhitagarwal1612/ground-android/pull/1/files

Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:302]: I like this better. Thanks for the suggestion! 
Would you like to open a PR with the proposed changes? If not, then I'd be more than happy to do it on your behalf. 😄

Comment (dturner) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:302]: I'm happy to do it. Will do shortly.

Comment (dturner) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:302]: https://github.com/google/ground-android/pull/529

**Pull request merged**